### ISSUE TITLE
Installation fails on python3.6 with .tar.gz package

### ISSUE THREAD

**Issue by mathPi** [shap/shap#1628]
Since the creation of numpy 1.20.0rc1, it is impossible to install shap on a linux environment.
This is because the version of numpy is not set in the setup.py file. So if numpy is not found, pip installs the latest numpy which requires python>=3.7.

This makes shap installation not working on python 3.6 when no wheels are available.
A workaround is to install a previous version of numpy before installing shap.

Steps to reproduce:
- make sure to clean the pip cache
- create a new empty virtual env
- """ pip install -v shap ""

Expected Result:
Shap must be compiled and installed (from .tar.gz package)

Observed result:
`
Running setup.py (path:/tmp/pip-wheel-fh2iiiel/shap_8a7e7a78d4bf4aec86b2fa60bd274ee2/setup.py) egg_info for package shap
    Created temporary directory: /tmp/pip-pip-egg-info-fm5rvl3w
    Running command python setup.py egg_info
    Python version >= 3.7 required.
    WARNING: The C extension could not be compiled, sklearn tree models not supported.
    Searching for numpy
    Reading https://pypi.org/simple/numpy/
    Downloading https://files.pythonhosted.org/packages/81/46/ecc1ea3e5819eb0d83306a8f468693785c94ed5467fcbb6f2529abf0102f/numpy-1.20.0rc1.zip#sha256=98f4e754f1c3db7ca53e53b1ef6474703b167af75f3784f99b1fe4dd936ea77f
    Best match: numpy 1.20.0rc1
    Processing numpy-1.20.0rc1.zip
    Writing /tmp/easy_install-q_cshyza/numpy-1.20.0rc1/setup.cfg
    Running numpy-1.20.0rc1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-q_cshyza/numpy-1.20.0rc1/egg-dist-tmp-6ipj6psy
    Python version >= 3.7 required.
    ERROR: Failed to build!
ERROR: No .egg-info directory found in /tmp/pip-pip-egg-info-fm5rvl3w
`

**Comment by slundberg** [comment]
Just to be clear. This fails because `pip install numpy` fails right?

**Comment by slundberg** [comment]
Perhaps the right solution would be to install numpy (a lower version) into the virtualenv first, then install SHAP. SHAP is just pulling the latest numpy because it was not already installed, it shouldn't upgrade a previously working lower version.

**Comment by mathPi** [comment]
Yes that's right, the numpy installation fails.
Actually this issue was a mistake from myself and from numpy pypi repo because the issue comes from numpy. The RC version is not pulled anymore for a python3.6. We can expect the same behavior for the next released version.
Thanks.

**Comment by mathPi** [comment]
I don't think that is the right solution because if you have python projects handling dependencies with dependency manager tools or even requirements.txt, it would mean to change the way dependencies are installed and complexify the process.

But there is something weird. I don't understand why the RC version of numpy is taken to build shap. Because an RC version is not stable by definition so must not be installed by default and this one is tagged to require python>=3.7 so it must not be selected for a python3.6.

**Comment by xGolden** [comment]
I agree with mathPi, when installing via requirements.txt, including a frozen numpy version prior to install shap does not resolve this issue

**Comment by asolimando** [comment]
It does not fix because, in my experience, `pip` does not necessarily follow the order of packages from `requirements.txt`.

The workaround for the moment in my py3.6 build is to install `numpy` and `scikit-learn` first, then the rest, as follows:
```
pip3 install --no-cache-dir numpy==1.19.5 scikit-learn==0.23.0 && \
 pip3 install --no-cache-dir -r requirements.txt
```

**Comment by xGolden** [comment]
Correct, pip will not necessarily follow the order of packages. This is compounded when deploying to an environment where you don't necessarily have complete control over build order (heroku, cloudfoundry).

The correct solution if desiring to continue support for Python 3.6 would be to detect python version and match the correct numpy version -- or provide a tag of shap for 3.6

**Comment by kbattocchi** [comment]
I believe this would be addressed by #1802

**Comment by connortann** [comment]
Closing as not planned, as python 3.6 is no longer supported,
